src/qengine/operators.cpp

```diff
@@ -931,6 +931,62 @@ void QEngineCPU::DIV(bitCapInt toDiv, bitLenInt inOutStart, bitLenInt carryStart
     }
 }
 
+void QEngineCPU::MULModNOut(bitCapInt toMul, bitLenInt inStart, bitLenInt outStart, bitLenInt length)
+{
+    SetReg(outStart, length, 0);
+
+    bitCapInt lowMask = (1U << length) - 1U;
+    toMul &= lowMask;
+    if (toMul == 0) {
+        return;
+    }
+
+    if (length > 0U) {
```

Comment:
Invert these if statements (here and elsewhere) to reduce the overall indentation depth of the code.

Reply:
I thought so, too.

Reply:
Changed. Really, there's no reason to check the `length == 0` case, so it has been removed.